tests/hardhat/XVS/XVSVaultFix.ts

```diff
@@ -0,0 +1,131 @@
+  await xvsVaultImpl.connect(admin)._become(xvsVaultProxy.address);
+  xvsVault = XVSVault__factory.connect(xvsVaultProxy.address, signer);
+}
+
+async function sendGasCost() {
+  /**
+   *  sending gas cost to owner
+   * */
+  [signer] = await ethers.getSigners();
+  console.log("-- Sending gas cost to owner addr --");
+  await signer.sendTransaction({
+    to: Owner,
+    value: ethers.BigNumber.from("900081987000000000"),
+    data: undefined,
+  });
+}
+
+describe("XVSVault", async () => {
+  if (FORK_MAINNET) {
+    it("Verify Expected Result old withdraw requests", async () => {
+      await reset(`https://wild-blissful-dawn.bsc.discover.quiknode.pro/${QUICK_NODE_KEY}`, 25458046);
+      await sendGasCost();
+      await deployAndConfigureVault();
+
+      const result = await xvsVault.getWithdrawalRequests(tokenAddress, poolId, affectedUserAddress);
+
+      expect(result[0][0].toString()).to.eql(beforeUpgrade[0][0].toString());
+      expect(result[0][1].toString()).to.eql(beforeUpgrade[0][1].toString());
+      expect(result[0][2].toString()).to.eql(beforeUpgrade[0][2].toString());
+
+      expect(result[1][0].toString()).to.eql(beforeUpgrade[1][0].toString());
+      expect(result[1][1].toString()).to.eql(beforeUpgrade[1][1].toString());
+      expect(result[1][2].toString()).to.eql(beforeUpgrade[1][2].toString());
+
+      expect(result[2][0].toString()).to.eql(beforeUpgrade[2][0].toString());
+      expect(result[2][1].toString()).to.eql(beforeUpgrade[2][1].toString());
+      expect(result[2][2].toString()).to.eql(beforeUpgrade[2][2].toString());
+
+      expect(result[3][0].toString()).to.eql(beforeUpgrade[3][0].toString());
+      expect(result[3][1].toString()).to.eql(beforeUpgrade[3][1].toString());
+      expect(result[3][2].toString()).to.eql(beforeUpgrade[3][2].toString());
+
+      expect(result[4][0].toString()).to.eql(beforeUpgrade[4][0].toString());
+      expect(result[4][1].toString()).to.eql(beforeUpgrade[4][1].toString());
+      expect(result[4][2].toString()).to.eql(beforeUpgrade[4][2].toString());
+    });
+
+    it("Verify Pending rewards of the user", async () => {
+      await reset(`https://wild-blissful-dawn.bsc.discover.quiknode.pro/${QUICK_NODE_KEY}`, 25458046);
+      await sendGasCost();
+      await deployAndConfigureVault();
+
+      const beforeUpgradeResult = await oldXVSVault.getUserInfo(tokenAddress, poolId, affectedUserAddress);
+      const afterUpgradeResult = await xvsVault.getUserInfo(tokenAddress, poolId, affectedUserAddress);
+
+      expect(beforeUpgradeResult[0].toString()).equals(afterUpgradeResult[0].toString());
+      expect(beforeUpgradeResult[1].toString()).equals(afterUpgradeResult[1].toString());
+      expect(beforeUpgradeResult[2].toString()).equals(afterUpgradeResult[2].toString());
+    });
+
+    it("Request Withdrawal Request", async () => {
+      await reset(`https://wild-blissful-dawn.bsc.discover.quiknode.pro/${QUICK_NODE_KEY}`, 25651479);
```

Comment:
Could we get this config parameter from the hardhat object? (to avoid hardcoded URL's in the code...)

Reply:
Done